Hakawai/Mentions/HKWMentionsPluginV2.m

```diff
@@ -1006,6 +1007,26 @@ - (BOOL)textView:(__unused UITextView *)textView shouldChangeTextInRange:(NSRang
         [self bleachMentionsIntersectingWithRange:range];
     }
 
+    // If we just did a non-programmatic paste, insert the text directly and add attributes
+    // This is needed because rich text formatting is off, which strips attribution from pasting, which we want to maintain.
+    // Once (if ever) we turn rich text formatting on, we can remove this.
+    if (self.wasNonProgrammaticPaste && text.length > 0) {
+        self.wasNonProgrammaticPaste = NO;
+        __strong __auto_type parentTextView = self.parentTextView;
+        [parentTextView transformTextAtRange:range withTransformer:^NSAttributedString *(__unused NSAttributedString *input) {
+            return [[NSAttributedString alloc] initWithString:text
+                                                   attributes:[self defaultTextAttributes]];
+        }];
+        parentTextView.selectedRange = NSMakeRange(range.location + [text length], 0);
+        returnValue = NO;
+
+        // Manually notify external delegate that the textView changed
+        id<HKWTextViewDelegate> externalDelegate = parentTextView.externalDelegate;
+        if ([externalDelegate respondsToSelector:@selector(textViewDidChange:)]) {
+            [externalDelegate textViewDidChange:parentTextView];
+        }
```

Comment:
why is this only needed for pasting

Reply:
explained on slack about that it's about overriding the shouldChangeText behavior